### PR TITLE
docs: add detail on spot interruption behavior

### PR DESCRIPTION
**Pull request by jhart0**
Fixes # 

**Description**

Doc only update adding additional detail on spot interruption behavior 

**How was this change tested?**

*

**Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note
Doc only update adding additional detail on spot interruption behavior 
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/concepts/deprovisioning.md [hunk 1]
```diff
@@ -138,6 +138,8 @@ data:
   ...
 ```
 
+For Spot Interruption Warnings, Karpenter will cordon and drain the node as per the above. There may be insufficient capacity to schedule the pods running on the affected node. The provisioner will then start a new instance to schedule these pods. Spot Interruptions have a 2 minute notice before Amazon EC2 reclaims the instance. Generally, this is sufficient time for the new node to become ready and to schedule the pods. Karpenter will follow the PDB while it drains the node. This means there is a chance that a node may not be fully drained by the time it is reclaimed.
```
Comment (jonathan-innis) [inline website/content/en/preview/concepts/deprovisioning.md:141]: Can you add this detail higher up where we say "When Karpenter detects..." at the end. I'm imagining the key details that we need to include here beyond what's currently there is that:
1. Karpenter spins up a new instance as soon as it sees the spot interruption
2. The spot interruption comes with a 2-minute warning, which is effectively the time that Karpenter has to spin up the new node
3. Mention that Karpenter's average node startup time means that this 2 minute startup is generally enough to spin up the new node and transfer the pods over.

**Review by jonathan-innis — COMMENTED** [review]
Thanks for the contribution! I think we can combine this with the detail on draining at the top! Good callouts! 🎉

File: website/content/en/preview/concepts/deprovisioning.md (hunk 1 above)
Comment (jhart0) [reply website/content/en/preview/concepts/deprovisioning.md]: Thanks @jonathan-innis  ! Addressed in latest commit

File: website/content/en/preview/concepts/deprovisioning.md [hunk 2]
```diff
@@ -118,6 +118,8 @@ If interruption-handling is enabled, Karpenter will watch for upcoming involunta
 
 When Karpenter detects one of these events will occur to your nodes, it automatically cordons, drains, and terminates the node(s) ahead of the interruption event to give the maximum amount of time for workload cleanup prior to compute disruption. This enables scenarios where the `terminationGracePeriod` for your workloads may be long or cleanup for your workloads is critical, and you want enough time to be able to gracefully clean-up your pods.
 
+For Spot interruptions, the provisioner will start a new instance as soon as it sees the Spot interruption warning. Spot interruptions have a __2 minute notice__ before Amazon EC2 reclaims the instance. Karpenter's average node startup time means that generally, this is sufficient time for the new node to become ready and to move the pods to the new node. Karpenter will respect the PDB while it drains the node. This means there is a chance that a node may not be fully drained by the time it is reclaimed.
```
Comment (jonathan-innis) [inline website/content/en/preview/concepts/deprovisioning.md:121]:
```suggestion
For Spot interruptions, the provisioner will start a new machine as soon as it sees the Spot interruption warning. Spot interruptions have a __2 minute notice__ before Amazon EC2 reclaims the instance. Karpenter's average node startup time means that, generally, there is sufficient time for the new node to become ready and to move the pods to the new node before the machine is reclaimed.
```

Small updates

**Review by jonathan-innis — COMMENTED** [review]
Just some wording and removal of some duplication, otherwise lgtm

File: website/content/en/preview/concepts/deprovisioning.md (hunk 2 above)
Comment (jhart0) [reply website/content/en/preview/concepts/deprovisioning.md]: cool, sounds good - just updated

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**